projects/plugins/jetpack/class.jetpack.php

```diff
@@ -1865,6 +1892,22 @@ public function check_open_graph() {
 			}
 		}
 
+		// Check if something has already rendered the OG tags.
+		$status = ob_get_status();
+		// Only end the output buffer if it's ours.
+		if ( ( $status['name'] ?? '' ) === get_class( $this ) . '::capture_head' ) {
+			ob_end_flush();
+		}
+		$processor = new WP_HTML_Tag_Processor( $this->captured_output );
+		while ( $processor->next_tag( 'meta' ) ) {
+			if ( ! empty( $processor->get_attribute( 'content' ) ) &&
+				str_starts_with( strtolower( $processor->get_attribute( 'property' ) ?? '' ), 'og:' ) ) {
```

Comment:
From the HTML API perspective, this looks good in general, I'll add a few notes.

The processor represents attributes as:
- `null`: attribute is absent
- `true`: attribute name is present with no value (common for booleans like `<input disabled>`).
- a string: the attribute has a value. It could be `""`.

I'd probably be more verbose in this to check the conditions. In particular, both attributes should have string values. I think it would look something like this (untested):

```suggestion
			$property = $processor->get_attribute( 'property' );
			if ( ! is_string( $property ) || ! str_starts_with( strtolower( $property ), 'og:' ) ) {
				continue;
			}
			
			// This is likely sufficient to check for content attribute
			if ( is_string( $processor->get_attribute( 'content' ) ) {
```

[There's also `substr_compare` that could be considered for the property value comparison.](https://www.php.net/manual/en/function.substr-compare.php)

Reply:
Gotcha! Thanks for looking at this @sirreal, that's some useful insight. You're right, I should be covering all the return types.